sim/priest/shadow/spriest.go

```diff
@@ -241,3 +237,318 @@ func (spriest *ShadowPriest) Act(sim *core.Simulation) time.Duration {
 		spriest.GetRemainingCD(core.GCDCooldownID, sim.CurrentTime),
 		spell.CastTime)
 }
+
+// CalculateMindflay will calculate how many ticks should be cast and mutate the cast.
+func (spriest *ShadowPriest) CalculateMindflay(sim *core.Simulation, spell *core.SimpleSpell, allCDs []time.Duration) time.Duration {
```

Comment:
Seems like this function should go in mind_flay.go

Reply:
this is mindflay rotation logic. We could rename it better if that would help..  "CalculateMindflayRotation"